cip/1.accepted/CIP2017-04-13-Aggregations.adoc

```diff
@@ -0,0 +1,110 @@
+= CIP2017-04-13 Aggregations
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Aggregations
+
+=== Syntax
+
+[source, ebnf]
+----
+Return = 'RETURN', ['DISTINCT'], ReturnBody, Filter ;
+With = 'WITH', ['DISTINCT'], ReturnBody, Filter ;
+SingleValueReturn = 'RETURN', (Expression | ProjectedMap | Aggregation), Filter ;
+
+ReturnBody = ('*' | ReturnItem), {',', ReturnItem} ;
+ReturnItem = (Expression, ['AS', Variable])
+           | (Aggregation, ['AS', Variable])
+           | (ProjectedMap, 'AS', Variable)
+           ;
+
+Filter = [Where], [Order], [Skip], [Limit] ;
+
+Aggregation = Aggregator, 'OF', Expression ;
+Aggregator = SymbolicName | ExtensionName ;
+ExtensionName = {SymbolicName, '.'}-, SymbolicName ;
+----
+
+=== Examples
+
+[source, cypher]
+.Aggregation using `avg`
+----
+MATCH (employee:Employee)
+RETURN avg OF employee.salary
+----
+
+[source, cypher]
+.Aggregation using `collect`
+----
+MATCH (person:Person)-[:FRIEND]-(friend)
+RETURN person.email, collect OF friend {.name,.email} AS friends
+----
+
+[source, cypher]
+.Aggregation using `count`
+----
+MATCH (nodes) RETURN count OF nodes
```

Comment:
Some similar ideas are discussed in the Python generator expression PEP -> https://www.python.org/dev/peps/pep-0289/